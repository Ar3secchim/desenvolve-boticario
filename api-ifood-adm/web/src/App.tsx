import React from 'react'
import { Routes, Route } from 'react-router-dom'

import Home from './Pages/Home'
import VitrineRestaurantes from './Pages/ShowcaseRestaurants'
import AdminRestaurantes from './Pages/AdminRestaurantes/AdminRestaurantes'
import FormularioRestaurante from './Pages/AdminRestaurantes/FormularioRestaurante'
import PageLayoutAdm from './Pages/AdminRestaurantes/PageLayoutAdm'

function App() {

	return (
		<Routes>
			<Route path="/" element={<Home />} />
			<Route path="/restaurantes" element={<VitrineRestaurantes />} />

			<Route path='/admin' element={<PageLayoutAdm />}>
				<Route path="restaurantes" element={<AdminRestaurantes />} />
				<Route path="restaurantes/novo" element={<FormularioRestaurante />} />
				<Route path="restaurantes/:id" element={<FormularioRestaurante />} />
			</Route>
		</Routes>
	)
}

export default App
