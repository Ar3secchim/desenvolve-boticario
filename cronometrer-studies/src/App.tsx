import { useState } from "react";

import style from "./app.module.scss"
import Form from "./components/form";
import List from "./components/list";
import Stopwatch from "./components/stopwatch";
import ITasks from "./types/tasks";

function App() {
  const [tasks, setTasks] = useState<ITasks[] | []>([])
  const [select, setSelect] = useState<ITasks>()

  function selectTasks(tasksSelect: ITasks){
    setSelect(tasksSelect)
    setTasks(tasksBefore => tasksBefore.map( task => ({
      ...task,
      select: task.id === tasksSelect.id ? true : false
    })))
  }  
  function finalyTask(){
    if(select){
      setSelect(undefined)
      setTasks(tasksbefore => tasksbefore.map(
        tasks => { if(tasks.id === select.id){
          return{
            ...tasks,
            select: false,
            check: true
          }
        }
        return tasks
      }))
    }
  }

  return (
    <div  className={style.AppStyle}>
      <div className={style.form}>
        <Form setTasks={setTasks} />
        <Stopwatch select={select} finalyTask={finalyTask}/>
      </div>
      
      <List 
        tasks={tasks}
        selecTasks ={selectTasks}  
      />

    </div>
  );
}

export default App
