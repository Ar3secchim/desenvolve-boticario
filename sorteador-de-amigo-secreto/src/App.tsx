import Footer from './componentes/Footer/Footer';
import Form from './componentes/Form/Formulario';
import { ListaParticipantes } from './componentes/ListaParticipantes/ListaParticipantes';

function App() {
  return (
   <>
    <Form />
    <ListaParticipantes />
    <Footer />
   </>
  )
}

export default App;
