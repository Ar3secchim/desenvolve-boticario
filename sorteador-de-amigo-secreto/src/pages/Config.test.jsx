import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import App from "../App";


const mockNavigation = jest.fn()

jest.mock('react-router-dom', ()=>{
  return{
    useNavigate: () => mockNavigation
  }
})

describe('a pagina de configuração',()=>{
  test('config pages',()=>{
    const {container} = render(
      <RecoilRoot>
        <App />
      </RecoilRoot>
    )
    expect(container).toMatchSnapshot()
  })
})