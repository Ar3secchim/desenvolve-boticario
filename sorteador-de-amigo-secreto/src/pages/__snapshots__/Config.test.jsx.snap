// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuração config pages 1`] = `
<div>
  <form>
    <input
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Adicionar
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Iniciar brincadeira
    </button>
  </footer>
</div>
`;
